test/jdk/sun/security/ssl/SignatureScheme/BlockSignatureSchemesForCert.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8366211
+ * @summary Block signature scheme names to be used with CertificateSignature
+ *          algorithm constraints usage
+ * @library /javax/net/ssl/templates
+ *          /test/lib
+ * @run main/othervm BlockSignatureSchemesForCert
+ */
+
+import static jdk.test.lib.Asserts.assertEquals;
+import static jdk.test.lib.Asserts.assertTrue;
+import static jdk.test.lib.Utils.runAndCheckException;
+
+import java.security.Security;
+
+public class BlockSignatureSchemesForCert extends SSLSocketTemplate {
+
+    public static void main(String[] args) throws Exception {
+        Security.setProperty("jdk.tls.disabledAlgorithms",
+                "rsa_pss_pss_sha256 usage CertificateSignature");
+
+        runAndCheckException(
+                () -> new BlockSignatureSchemesForCert().run(),
+                e -> {
+                    assertTrue(e instanceof ExceptionInInitializerError);
```

Comment:
nit: Do you think this might be a bit easier to read? 

```suggestion
                    assertTrue(e instanceof ExceptionInInitializerError);
                    assertTrue(e.getCause() instanceof IllegalArgumentException);
                    assertEquals(e.getCause().getMessage(),
                            "Can't use signature scheme names with "
                            + "CertificateSignature usage constraint");
```